conda_smithy/ci_register.py

```diff
@@ -192,17 +192,36 @@ def add_project_to_travis(user, project):
             if "repo" in content:
                 break
         except requests.HTTPError as e:
-            # We regularly seem to hit this issue during automated feedstock registration on github.
-            # https://github.com/conda-forge/conda-smithy/issues/233
-            # Repo is not available on travis-ci yet.
             print(e)
 
         if count == 1:
+            def wait_until_synced(ignore=False):
+                is_sync_url = '{}/users'.format(endpoint)
+                response = requests.post(is_sync_url, headers=headers)
+                content = response.json()
+                for c in range(20):
+                    if ("is_syncing" in content and content["is_syncing"] == False):
+                        break
+                    time.sleep(3)
+                else:
+                    if ignore:
+                        print(" * Travis is being synced by somebody else. Ignoring")
+                    else:
+                        raise RuntimeError('Synching is not finishing for a minute now')
+
+            wait_until_synced(ignore=True)
```

Comment:
So I'm not totally sure about this. As it is now in staged-recipes, we create all the repos on GitHub and then we register CIs. Thus the first sync, even though it is quite long running, eventally correctly handles all of the repos that were just added. This sort of change would cause there to be a sync for each repo that is being added. I'm a little concerned about adverse effects from so many syncs back-to-back. Maybe we should recheck after the sync to see if the repo is now included and skip the syncing if so.